spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/DistriOptimizer.scala

```diff
@@ -212,13 +218,21 @@ object DistriOptimizer extends AbstractOptimizer {
         .zipPartitions(models, preservesPartitioning = true) { (data, modelIter) => {
           val cached = modelIter.next()
           val syWStart = System.nanoTime()
+          if (isWeightUpdateDeferred && localGradientBuffer == null) {
+            localGradientBuffer = Tensor[T](parameters.size)
+          }
+          val gradientBuffer = localGradientBuffer.asInstanceOf[Tensor[T]]
           /*
             Note: All models in `cached` share the same storage for weights, so we only need to
             copy the weights from parameter server into the first model's weights.
            */
-          val weightsResults = parameters.values.map(p =>
-            p.getWeights(cached.modelWeights.head.narrow(1, p.paramOffset, p.size))
-          ).toArray
+          val weightsResults = if (!isWeightUpdateDeferred || iteration == 0) {
```

Comment:
What will happen if we resume a training?

Reply:
The updated code can fall back to fetching global weights if the local model cache does not have the updated weights